### PR TITLE
Add noexec support to sudoers

### PR DESCRIPTION
**Pull request by adaniaud**
##### SUMMARY

This adds support for the NOEXEC tag in the sudoers module

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

sudoers

##### ADDITIONAL INFORMATION

This add support to the NOEXEC tag of the sudoers configuration file to, which prevents an executable from running another command itself.

https://www.sudo.ws/docs/man/1.8.32/sudoers.man/#NOEXEC

```
- name: Create rule with default parameters
  community.general.sudoers:
    name: my-sudo-rule-default
    state: present
    user: alice
    commands: /usr/bin/less

- name: Create rule with noexec parameters
  community.general.sudoers:
    name: my-sudo-rule-noexec
    state: present
    user: alice
    commands: /usr/bin/less
    noexec: true


TASK [sudoers : Create rule with default parameters] ***************************
changed: [testhost] => {"changed": true}

TASK [sudoers : Create rule with noexec parameters] ****************************
changed: [testhost] => {"changed": true}


root@ff22fb1bc706:/etc/sudoers.d$ cat my-sudo-rule-default         
alice ALL=NOPASSWD: /usr/bin/less

root@ff22fb1bc706:/etc/sudoers.d$ cat my-sudo-rule-noexec
alice ALL=NOEXEC:NOPASSWD: /usr/bin/less
```

### CODE REVIEW

File: changelogs/fragments/7983-sudoers-add-support-noexec.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - sudoers - add support for the NOEXEC tag in sudoers rules (https://github.com/ansible-collections/community.general/pull/7983).
```
Comment (felixfontein) [inline changelogs/fragments/7983-sudoers-add-support-noexec.yml:2]:
```suggestion
  - sudoers - add support for the ``NOEXEC`` tag in sudoers rules (https://github.com/ansible-collections/community.general/pull/7983).
```

File: plugins/modules/sudoers.py [hunk 2]
```diff
@@ -45,6 +45,11 @@
       - The name of the sudoers rule.
       - This will be used for the filename for the sudoers file managed by this rule.
     type: str
+  noexec:
+    description:
+      - Whether a command is prevented to run further commands itself.
+    default: false
+    type: bool
```
Comment (felixfontein) [inline plugins/modules/sudoers.py:52]:
```suggestion
    type: bool
    version_added: 8.4.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

**Comment by adaniaud** [comment]
Thanks for your review, I added your suggestions

**Review by felixfontein — APPROVED** [review]
Looks good to me. If nobody objects, I'll merge this on the weekend.

**Comment by JonEllis** [comment]
Yeah, this seems good to me. While I've not used the noexec feature myself, it's usage in this PR seem consistent with the docs, and generated suders files should normally be checked by visudo anyway as part of using the module.

Thanks for your contribution!

**Comment by felixfontein** [comment]
@adaniaud thanks for your contribution!
@JonEllis thanks for reviewing this!

**Comment by adaniaud** [comment]
It's a small contribution but hopefully will be useful to some.
Thank you guys for reviewing it and of course thank you for all your work
as maintainers!

**Pull request merged**